### DISCUSSION TITLE
How to I set up a watch folder in Papermerge 3.0.2?

### DISCUSSION THREAD

**Discussion by andersonlyle** [ciur/papermerge#610] [General]
I've successfully installed Papermerge on my Synology NAS. However, I can't find any instructions on how to set up a watch folder that I can drop PDFs into to load into Papermerge.  Can anyone point me in the right direction?  Thanks!

**Comment by ciur** [comment]
There is no watcher feature as part of application itself.

What you can do, is to write some sort of minimal watcher as separate application which will upload documents to the Papermerge instance via REST API.
Or you can use something (I don't know bash script ?) which will check content of the folder in a loop and invoke (via bash?) [papermerge-cli](https://github.com/papermerge/papermerge-cli) for uploading files (papermerge-cli uses REST API for that).

In any case, the idea is to use external app which will import documents into Papermerge instance via REST API.
You may ask why ?
The answer is: to avoid ever growing monolithic behemoth doing everything (import from folder, watcher, import from S3, import from google drive etc).

The REST API is there. What is missing is the watcher.
